File: la-palma-experience/analyze-pois.js
// Script to analyze POI database
const fs = require('fs');
const path = require('path');

// Import places data
const placesPath = path.join(__dirname, 'src/data/places.js');
const placesContent = fs.readFileSync(placesPath, 'utf8');

// Parse the places array (simple extraction)
const placesMatch = placesContent.match(/export const places = \[([\s\S]*)\];/);
if (!placesMatch) {
  console.log('Could not parse places data');
  process.exit(1);
}

// Use eval to parse the array (in a controlled environment)
const places = eval('[' + placesMatch[1] + ']');

console.log('=== POI DATABASE ANALYSIS ===\n');
console.log(`Total POIs: ${places.length}\n`);

// Count by type
const byType = {};
places.forEach(p => {
  byType[p.type] = (byType[p.type] || 0) + 1;
});

console.log('POIs by type:');
Object.entries(byType).sort((a, b) => b[1] - a[1]).forEach(([type, count]) => {
  console.log(`  ${type}: ${count}`);
});

console.log('\n=== IMAGE ANALYSIS ===\n');

// Check images
const localImages = places.filter(p => p.image && p.image.startsWith('/images/'));
const unsplashImages = places.filter(p => p.image && p.image.includes('unsplash.com'));
const noImage = places.filter(p => !p.image);

console.log(`POIs with local images: ${localImages.length}`);
console.log(`POIs with Unsplash images: ${unsplashImages.length}`);
console.log(`POIs without images: ${noImage.length}\n`);

console.log('POIs using Unsplash (need local images):');
unsplashImages.forEach(p => {
  console.log(`  ID ${p.id}: ${p.name} (${p.type})`);
});

console.log('\n=== DUPLICATE DETECTION ===\n');

// Check for duplicate names
const nameMap = {};
places.forEach(p => {
  if (!nameMap[p.name]) {
    nameMap[p.name] = [];
  }
  nameMap[p.name].push(p.id);
});

const duplicates = Object.entries(nameMap).filter(([name, ids]) => ids.length > 1);
console.log(`Potential duplicates found: ${duplicates.length}\n`);
duplicates.forEach(([name, ids]) => {
  console.log(`  "${name}": IDs ${ids.join(', ')}`);
});

console.log('\n=== DESCRIPTION QUALITY ===\n');

// Check description length
const shortDescriptions = places.filter(p => p.description && p.description.length < 200);
console.log(`POIs with short descriptions (<200 chars): ${shortDescriptions.length}`);
shortDescriptions.forEach(p => {
  console.log(`  ID ${p.id}: ${p.name} - ${p.description.length} chars`);
});

console.log('\n=== AVAILABLE LOCAL IMAGES ===\n');

// Check available images in public/images
const imagesDir = path.join(__dirname, 'public/images');
if (fs.existsSync(imagesDir)) {
  const imageFiles = fs.readdirSync(imagesDir);
  console.log(`Total image files available: ${imageFiles.length}\n`);
  imageFiles.forEach(file => {
    const used = places.find(p => p.image === `/images/${file}`);
    console.log(`  ${file} - ${used ? `✓ Used by "${used.name}"` : '✗ NOT USED'}`);
  });
}

console.log('\n=== POI LIST BY TYPE ===\n');

Object.keys(byType).sort().forEach(type => {
  console.log(`\n${type.toUpperCase()}:`);
  const poisOfType = places.filter(p => p.type === type);
  poisOfType.forEach(p => {
    const imageStatus = p.image ? (p.image.startsWith('/images/') ? '✓' : '⚠') : '✗';
    const descLen = p.description ? p.description.length : 0;
    console.log(`  ${imageStatus} ID ${p.id}: ${p.name} (${descLen} chars)`);
  });
});

File: la-palma-experience/analyze_pois.py
#!/usr/bin/env python3
import os
import re
import json

# Read the places.js file
places_path = '/Users/andreapiani/Library/Mobile Documents/com~apple~CloudDocs/Prototipi/insieme/la-palma-experience/src/data/places.js'

with open(places_path, 'r', encoding='utf-8') as f:
    content = f.read()

# Extract POI objects using regex
poi_pattern = r'\{\s*id:\s*(\d+),\s*name:\s*"([^"]+)",\s*type:\s*"([^"]+)"[^}]*image:\s*"([^"]*)"'
matches = re.findall(poi_pattern, content)

print('=== POI DATABASE ANALYSIS ===\n')
print(f'Total POIs found: {len(matches)}\n')

# Organize by type
by_type = {}
for poi_id, name, poi_type, image in matches:
    if poi_type not in by_type:
        by_type[poi_type] = []
    by_type[poi_type].append({
        'id': int(poi_id),
        'name': name,
        'image': image
    })

print('POIs by type:')
for poi_type in sorted(by_type.keys()):
    print(f'  {poi_type}: {len(by_type[poi_type])}')

print('\n=== IMAGE ANALYSIS ===\n')

local_images = []
unsplash_images = []
no_image = []

for poi_id, name, poi_type, image in matches:
    if image.startswith('/images/'):
        local_images.append((poi_id, name, poi_type, image))
    elif 'unsplash.com' in image:
        unsplash_images.append((poi_id, name, poi_type))
    else:
        no_image.append((poi_id, name, poi_type))

print(f'POIs with local images: {len(local_images)}')
print(f'POIs with Unsplash images: {len(unsplash_images)}')
print(f'POIs without images: {len(no_image)}\n')

print('POIs using Unsplash (need local images):')
for poi_id, name, poi_type in unsplash_images:
    print(f'  ID {poi_id}: {name} ({poi_type})')

print('\n=== DUPLICATE NAMES ===\n')

# Check for duplicate names
name_map = {}
for poi_id, name, poi_type, image in matches:
    if name not in name_map:
        name_map[name] = []
    name_map[name].append(int(poi_id))

duplicates = {name: ids for name, ids in name_map.items() if len(ids) > 1}
print(f'Potential duplicates found: {len(duplicates)}\n')
for name, ids in sorted(duplicates.items()):
    print(f'  "{name}": IDs {ids}')

print('\n=== AVAILABLE LOCAL IMAGES ===\n')

images_dir = '/Users/andreapiani/Library/Mobile Documents/com~apple~CloudDocs/Prototipi/insieme/la-palma-experience/public/images'
if os.path.exists(images_dir):
    image_files = sorted([f for f in os.listdir(images_dir) if f.endswith(('.jpg', '.jpeg', '.png', '.webp'))])
    print(f'Total image files available: {len(image_files)}\n')

    used_images = {image.split('/')[-1] for _, _, _, image in matches if image.startswith('/images/')}

    for file in image_files:
        status = '✓ USED' if file in used_images else '✗ NOT USED'
        # Find which POI uses it
        poi_using = None
        for poi_id, name, poi_type, image in matches:
            if image.endswith(file):
                poi_using = f'by "{name}" (ID {poi_id})'
                break

        print(f'  {file} - {status} {poi_using if poi_using else ""}')

print('\n=== POIS BY TYPE (with image status) ===\n')

for poi_type in sorted(by_type.keys()):
    print(f'\n{poi_type.upper()}:')
    for poi in sorted(by_type[poi_type], key=lambda x: x['id']):
        if poi['image'].startswith('/images/'):
            status = '✓'
        elif 'unsplash.com' in poi['image']:
            status = '⚠'
        else:
            status = '✗'
        print(f'  {status} ID {poi["id"]}: {poi["name"]}')
